tests/protocol/mocks.py

```diff
@@ -101,15 +97,17 @@ def http_request(method, url, data, **kwargs):
                 return return_value
 
         # if the request was not handled try to use the mock wireserver data
-        if _USE_MOCK_WIRE_DATA_RE.match(url) is not None:
```

Comment:
When I designed the mock protocol I wanted requests that would return mock data to be explicit (they had to start with http://mock-goal-state (or https)). As I've been adding more data files this becomes cumbersome. Furthermore, the underlying implementation of the mocks in mockwiredata.py always try to use mock data.

While working on this PR I forgot to mark my mock data with "mock-goal-state" and all the new unit tests were ok, but some existing tests were failing. Turns out that the test request was failing on the direct call and was falling back to the HostGAPlugin, changing the default channel in the process.

Some of the HostGAPlugin were assuming the initial state of the default channel, but that is incorrect, since it is actually a static variable that can be changed as a side effect of other test.

I fixed the HostGAPlugin tests, but I also dropped the enforcement of "mock-goal-state" in the test data. Not only the test data is easier to write, but now the behaviour of this mock is consistent with the behaviour of the mock HostGAPlugin (both of them always try to use mock data)